nestkernel/eprop_archiving_node_impl.h

```diff
@@ -138,14 +150,40 @@ EpropArchivingNode< HistEntryT >::erase_used_eprop_history()
     }
     else
     {
-      const auto it_eprop_hist_from = get_eprop_history( t );
-      const auto it_eprop_hist_to = get_eprop_history( t + update_interval );
-      eprop_history_.erase( it_eprop_hist_from, it_eprop_hist_to ); // erase found entries since no longer used
+      const auto it_eprop_hist_from_1 = get_eprop_history( t );
+      const auto it_eprop_hist_to_1 = get_eprop_history( t + update_interval );
+      eprop_history_.erase( it_eprop_hist_from_1, it_eprop_hist_to_1 ); // erase found entries since no longer used
     }
   }
-  const auto it_eprop_hist_from = get_eprop_history( 0 );
-  const auto it_eprop_hist_to = get_eprop_history( update_history_.begin()->t_ );
-  eprop_history_.erase( it_eprop_hist_from, it_eprop_hist_to ); // erase found entries since no longer used
+  const auto it_eprop_hist_from_2 = get_eprop_history( 0 );
+  const auto it_eprop_hist_to_2 = get_eprop_history( update_history_.begin()->t_ );
+  eprop_history_.erase( it_eprop_hist_from_2, it_eprop_hist_to_2 ); // erase found entries since no longer used
+}
+
+template < typename HistEntryT >
+void
+EpropArchivingNode< HistEntryT >::erase_used_eprop_history( const long eprop_isi_trace_cutoff )
+{
+  if ( eprop_history_.empty()     // nothing to remove
+    or update_history_.size() < 2 // no time markers to check
+  )
+  {
+    return;
+  }
+
```

Comment:
Provide comments explaining which parts of the history you are removing here and on lines 184ff below.

Reply:
Fixed in [36c9ec9](https://github.com/nest/nest-simulator/pull/3207/commits/36c9ec9ef229a772f74079b8f5740380b7735d03)